snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -45,6 +45,8 @@
 logger = logging.getLogger(__name__)
 
 _NETWORK_PROBE_COMMAND = \
+    'import sys; sys.excepthook = lambda t,e,b : ' \
+    'print("{{}}: {{}}".format(t.__name__, e));' \
```

Comment:
excepthook feels like overkill here. Why not just a try/except?

Reply:
The lack of line breaks is deliberate here because this is being run with `python -c`. As such we can't use try/except because that requires a line break.
If we wanted this to be more beautiful we'd have to have a script that we push into the container after it's created and run instead of the string. To my mind that's not worth it. Unless others feel strongly about pretty code.

Reply:
You've already had to modify this code. I think it's important to make it as maintainable as possible, for the same reason we want other code to be maintainable. In addition, you simply can't accomplish everything in a single line. Try running this:

```
import subprocess
import textwrap

_CODE = "import textwrap; for i in range(2): print('hello')"

subprocess.check_call(['python3', '-c', _CODE])
```

It doesn't work. However, this does, is just as easy, and looks way nicer:

```
import subprocess
import textwrap

_CODE = textwrap.dedent("""\
        import textwrap

        for i in range(2):
                print('hello')
""")

subprocess.check_call(['python3', '-c', _CODE])
```

Note that this works because the line breaks are preserved, but it's still just a string that you're handing to `-c` like normal.